utils/src/main/java/org/jboss/windup/util/xml/XmlUtil.java

```diff
@@ -31,7 +31,8 @@
 
 public class XmlUtil
 {
-    private static Logger LOG = Logging.get(XmlUtil.class);
+	private static final XPathFactory XPATH_FACTORY = XPathFactory.newInstance();
```

Comment:
Couple of notes: 

You could wrap with ThreadLocal. 

xml processing isn't multithreaded in our implementation today (so far as I can tell). 

This approach is used elsewhere in our code base and would need to be updated there too when we settle on an approach. 

Caching this did cut down on execution time in my test. 



> On Apr 24, 2015, at 11:34 PM, Jesse Sightler <notifications@github.com> wrote:
> 
> In utils/src/main/java/org/jboss/windup/util/xml/XmlUtil.java:
> 
> > @@ -31,7 +31,8 @@
> >  
> >  public class XmlUtil
> >  {
> > -    private static Logger LOG = Logging.get(XmlUtil.class);
> > +	private static final XPathFactory XPATH_FACTORY = XPathFactory.newInstance();
> XPathFactory is not thread safe:
> http://docs.oracle.com/javase/7/docs/api/javax/xml/xpath/XPathFactory.html
> 
> The code that uses this needs to be synchronized in order to use this pattern.
> 
> —
> Reply to this email directly or view it on GitHub.
>